### PR TITLE
Fill in x/ypixel in struct winsize for the TIOCGWINSZ ioctl.

### PR DESCRIPTION
**Pull request by 0x09**
https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man4/tty.4.html

iTerm currently fills in the row/column fields, but not the pixel density. This commit brings it in line with Terminal's behavior.

I was actually using this in combination with terminfo commands to play gifs before gif support was added.

### CODE REVIEW

File: sources/PTYTask.m [hunk 1]
```diff
@@ -43,6 +43,8 @@
                 struct winsize* win,
                 int width,
                 int height,
+                int pixelWidth,
```
Comment (gnachman) [inline sources/PTYTask.m]: Should be an `unsigned short` to match the fields in the winsize struct.

File: sources/PTYSession.m [hunk 2]
```diff
@@ -1169,7 +1169,7 @@ - (void)setWidth:(int)width height:(int)height
 {
     DLog(@"Set session %@ to %dx%d", self, width, height);
     [_screen resizeWidth:width height:height];
-    [_shell setWidth:width height:height];
+    [_shell setWidth:width height:height pixelWidth:width*_textview.charWidth pixelHeight:height*_textview.lineHeight];
```
Comment (gnachman) [inline sources/PTYSession.m]: Please put spaces around the `*` operators.

**Comment by gnachman** [comment]
The tty should be notified when the session size changes because of a font change that does not adjust the session's row/col size. I'd do it in this method:
````
- (void)setFont:(NSFont*)font
     nonAsciiFont:(NSFont*)nonAsciiFont
    horizontalSpacing:(float)horizontalSpacing
    verticalSpacing:(float)verticalSpacing
````

at the end of this if statement's block:
````
        if ([iTermPreferences boolForKey:kPreferenceKeyAdjustWindowForFontSizeChange]) {
            [[[self tab] parentWindow] fitWindowToTab:[self tab]];
````

File: sources/PTYTask.m (hunk 1 above)
Comment (0x09) [inline sources/PTYTask.m]: Shall I apply this to the existing width/height params? They aren't used inside these functions besides to assign ws_row/col.
Same regarding `launchWithPath` and `setWidth:...`

Comment (gnachman) [inline sources/PTYTask.m]: If you're feeling ambitious, yes please! This code is old and crufty and I like to clean it up as I go, but I won't hold up this PR for that.

**Comment by shabbyrobe** [comment]
This would help me a lot; is there anything specific (beyond a bit of a merge/rebase) that needs to happen to resurrect this PR?

**Comment by gnachman** [comment]
Implemented in commit ccf7ed6cd

**Comment by shabbyrobe** [comment]
Amazing! Thank you so much!

**Pull request closed without merging**